### PR TITLE
HOTT-3514: Add CI jobs to deploy Terraform

### PR DESCRIPTION
**Pull request by amberstarlight**
### Jira link

[HOTT-3514](https://transformuk.atlassian.net/browse/HOTT-3514)

### What?

I have added/removed/altered:

- Heavily modifed the CircleCi `config.yml`:
  - Removed jobs/commands that have been ported across to the `trade-tariff` Orb
  - Added `terraform` jobs

### Why?

I am doing this because:

- The time has come where we can start to deploy this app to the new AWS environment.

### Notes

So far I've been adding `[ci skip]` to all my pushes on this branch, so as to avoid redeploying the app several times. I'll want to do a run on this before we merge to see what the pipeline looks like from a branch that isn't main.

It's likely that we actually want to make changes to this - as in, we may want to deploy over development for every PR, but I've had my eyes full of YAML for the past three days and I can no longer think about them in great detail without a break from it.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -447,80 +398,74 @@ jobs:
...
-    steps:
-      - cf_purge_reviews
-
   deploy_main_to_staging:
-    docker:
-      - image: cimg/ruby:3.2.2-node
+    executor:
+      name: tariff/ruby
     environment:
-      SENTRY_ENVIRONMENT: "staging"
```
Comment (willfish) [inline .circleci/config.yml:490]: Assuming this should remain staging?

Comment (amberstarlight) [reply .circleci/config.yml:490]: Yep, looks like I did some copypasta

File: .circleci/config.yml [hunk 2]
```diff
@@ -2,98 +2,76 @@ version: 2.1
...
-            cf delete "$CF_APP-<< parameters.environment_key >>" -f
-
-            # Switch name of "dark" version to claim correct name
-            cf rename "$CF_APP-<< parameters.environment_key >>-dark" "$CF_APP-<< parameters.environment_key >>"
+            cf map-route \
+              "$CF_APP-<< parameters.environment_key >>-dark" \
+              london.cloudapps.digital \
+              -n "$CF_APP-<< parameters.environment_key >>-dark"
+
+            cf add-network-policy "$CF_APP-<< parameters.environment_key >>-dark" \
```
Comment (willfish) [inline .circleci/config.yml]: Any thoughts on splitting the dark app name onto a new line?

File: .circleci/config.yml [hunk 3]
```diff
@@ -447,80 +423,74 @@ jobs:
...
-    environment:
-      SENTRY_ENVIRONMENT: "development"
-    steps:
-      - cf_purge_reviews
-
   deploy_main_to_staging:
-    docker:
-      - image: cimg/ruby:3.2.2-node
+    executor:
+      name: tariff/ruby
```
Comment (willfish) [inline .circleci/config.yml:460]: Should this be a ruby-node executor?

File: .circleci/config.yml [hunk 4]
```diff
@@ -447,80 +423,74 @@ jobs:
...
+          environment_key: staging
+          app_domain_prefix: staging
+      - tariff/sentry-release:
+          environment: staging
 
   deploy_release_to_staging:
-    docker:
-      - image: cimg/ruby:3.2.2-node
+    executor:
+      name: tariff/ruby
```
Comment (willfish) [inline .circleci/config.yml:478]: Same again re ruby-node executor

File: .circleci/config.yml (hunk 3 above)
Comment (amberstarlight) [reply .circleci/config.yml:460]: So this uses the orb `ruby` executor, which [defaults](https://github.com/trade-tariff/trade-tariff-ci-orb/blob/main/src/executors/ruby.yml) to `3.2.2-node`.

File: .circleci/config.yml (hunk 4 above)
Comment (amberstarlight) [reply .circleci/config.yml:478]: As above, so below 🙏🏻

File: .circleci/config.yml (hunk 3 above)
Comment (willfish) [reply .circleci/config.yml:460]: Oh. I guess the inverse of this is that when we only want a smaller ruby image we end up with node as well?

File: .circleci/config.yml (hunk 2 above)
Comment (amberstarlight) [reply .circleci/config.yml]: A good idea! Done in https://github.com/trade-tariff/trade-tariff-frontend/pull/1515/commits/748a017e06ee67a24a7ea9968f1015a6b5beb093 :)

Comment (willfish) [reply .circleci/config.yml]: :tada:

**Review by willfish — APPROVED** [review]

**Pull request merged**